### PR TITLE
rockstar_games_launcher: new app

### PR DESCRIPTION
**Pull request by Klairm**
uwu

### CODE REVIEW

**Comment by Kreyren** [comment]
Please make a merge request in `kreytricks:upstream` that is a branch synced with upstream assuming that we will be pushing in there.

**Comment by Kreyren** [comment]
What is the status on this? Did you made the function for GTA V yet?

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Provide reference

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: `w_override_dlls native dinput8` is already set by `w_call dinput8` what is the reasoning for builtin?

**Comment by Klairm** [comment]
We have rockstar-games-launcher.exe which extracts files in C:/Program Files/rockstarsomethibg and which doesnt work with condlg32 provides with wine so it wont extract the files and permanent fix for that hss to be made in wine

since that is the case we are using 7z to skip this step

then the extracted files has Launcher.exe which is important for us since its the launcher we need to get authentication for rockstar GTA V

we know that Launcher. exe needs at least vcrun2017 (it has ms-*.dll libs in its root ehich are vcrun2017 libs) and dotnet452 (which shouldn't be required)
Also the Launcher requires at least win7 to be launched, and even setting winver to 7 or 10 it will still saying that the version is not supported, weird.

EDIT: Got the launcher working, and GTA V also, but i will need DXVK because now it looks very buggy

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: We need a reference here

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Can we use checksum or is this being updated to often?

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Please make a 'trigger' for this.

i.e: If directory is not present -> Create it

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Make a new comment in github repository and attempt to make a winebug-like report, once i approve it you will submit it in wine bugzilla if this bug is not known already

again **Do not submit bugs without my approval** i want to make sure that we make it as clear as possible for wine devs to know what is the issue

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: This requires winebug reference and comment above with reasoning,

For now add TODO comment above so that we won't forget about this

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Kreytricks specific: Make a trigger for this.

i.e: If package is not extracted -> Extract it (ideally use w_debug() for both scenarios)

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: What about this? The rockstar_games_launcher should work without us removing these libraries right?

If not change 7z to not extract these

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Use `w_override_app_winenv` instead so that we are not changing wineprefix-wide winenv

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Provide reasoning for this usage in the code with some reference

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Do we really need dxvk096 for rockstar_games_launcher?

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Why is this here?

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: There should be a space, follow the style other verbs are written

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Provide `installed_exe1` too

in kreytricks this is used to generate a shortcut

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Still this is not sufficient, Why is it needed for rockstar_game_launcher?

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: i meant space between `# ------` and `}` :p

Example:
image

**Comment by Klairm** [comment]
Now the rockstar games launcher it's fully working. 

Note that the first launch it will crash, but just ignore it and run it again.

**Comment by Kreyren** [comment]
@Klairm Can't merge, please resolve the review in file changes.

**Review by Kreyren — CHANGES_REQUESTED** [review]
@Klairm Can't merge, please resolve the review in file changes.

File: src/winetricks
Comment (Klairm) [reply src/winetricks]: It will be updated many times

File: src/winetricks
Comment (Klairm) [reply src/winetricks]: Checked that, dinput8 not really required

File: src/winetricks
Comment (Klairm) [reply src/winetricks]: Done

File: src/winetricks
Comment (Klairm) [reply src/winetricks]: I don't see it on my code O.o

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: This is wrong, it expects windows path not unix!

File: src/winetricks
Comment (Kreyren) [reply src/winetricks]: @Klairm Provide a comment informing about this fact then

File: src/winetricks
Comment (Kreyren) [reply src/winetricks]: but i can :p try squashing mby?

File: src/winetricks
Comment (Klairm) [reply src/winetricks]: The Rockstar Games Launcher doesn't support 32 bit system -> https://support.rockstargames.com/articles/360035473314/Rockstar-Games-Launcher-system-requirements

**Comment by Kreyren** [comment]
TODO: verify compatiblity on 32-bit

(this is for me since you cant test this)

File: src/winetricks
Comment (Kreyren) [inline src/winetricks]: Kreytricks specific: Set this per wineapp

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15322]: Kreytricks specific: use printf instead of echo (https://unix.stackexchange.com/a/65819)

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15341]: we dont need an if statement if this sort, but vcrun libs seems to be exported with wine so we should make a tracking for missing features

referencing : https://appdb.winehq.org/objectManager.php?sClass=application&iId=5766 (notice the note)

So make a winebug specifying that rockstar launcher needs foo.dll and attach a WINEDEBUG="-all, +loaddll" so that we can see which libs are called

EDIT: cross-reference the bug with this merge request

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15311]: there should be a newline after installed_exe1 and above load_...

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15305]: there should be a new line above w_metadata

**Comment by Kreyren** [comment]
@Klairm This is taking too long so I'm giving you dead line: Nov 11 2019 15:00 CET

if you need a help then ask me

Update: Delayed based on conversation with contributor

File: src/winetricks
Comment (Kreyren) [reply src/winetricks]: Unresolved: this is not resolved

**Comment by Kreyren** [comment]
@Klairm since i'm mentoring you to contribute in winetricks i've added `Kreytricks specific:` for issues that are kreytricks specific (since kreytricks is oversanitized and has different method to write code)

So i will merge this if you resolve only the issues that are not prefixed with `Kreytricks specific: `

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15314]: This newline shoudn't be there

File: src/winetricks
Comment (Kreyren) [reply src/winetricks]: Unresolved: this is not resolved

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15330]: wrong indentation

**Comment by Kreyren** [comment]
Affected by https://bugs.winehq.org/show_bug.cgi?id=48099

**Comment by Kreyren** [comment]
Affected by https://bugs.winehq.org/show_bug.cgi?id=47794

**Comment by Kreyren** [comment]
Affected by https://bugs.winehq.org/show_bug.cgi?id=47912

**Comment by Kreyren** [comment]
Tracking for affected bugs

https://bugs.winehq.org/buglist.cgi?bug_status=UNCONFIRMED&bug_status=NEW&bug_status=ASSIGNED&bug_status=STAGED&bug_status=REOPENED&bug_status=NEEDINFO&bug_status=RESOLVED&bug_status=CLOSED&list_id=679362&query_format=advanced&resolution=---&resolution=FIXED&resolution=INVALID&resolution=WONTFIX&resolution=DUPLICATE&resolution=WORKSFORME&resolution=ABANDONED&resolution=NOTOURBUG&short_desc=rockstar%20games%20launcher&short_desc_type=allwords

File: src/winetricks
Comment (Kreyren) [inline src/winetricks:15356]: There should be a new-line separating w_metadata and #----...

**Review by Kreyren — APPROVED** [review]
good enough for the beginner

**Pull request merged**
